cfe_internal/enterprise/CFE_hub_specific.cf

```diff
@@ -91,6 +99,43 @@ bundle common cfe_internal_hub_vars
 
 }
 
+##################################################################
+#
+# update_cli_rest_server_url_config
+#  - updates REST server URL port of Mission Portal WebGUI when
+#    cfe_internal_hub_vars.https_port is changed
+#
+##################################################################
+bundle agent update_cli_rest_server_url_config
+{
+  vars:
+    "mp_config_file" string => "$(cfe_internal_hub_vars.docroot)/application/config/config.php";
+    "mp_share_config_file" string => "$(sys.workdir)/share/GUI/application/config/config.php";
+    "regex_test_pattern" string => ".*localhost:$(cfe_internal_hub_vars.https_port).*";
+
+  files:
+    "$(mp_config_file)"
+      edit_line => change_cli_rest_server_url_port,
+      if => and(
+        fileexists("$(mp_config_file)"),
+        islessthan(countlinesmatching("$(regex_test_pattern)", "$(mp_config_file)"), 1)
+        );
+   "$(mp_share_config_file)"
+      edit_line => change_cli_rest_server_url_port,
+      if => and(
+        fileexists("$(mp_share_config_file)"),
+        islessthan(countlinesmatching("$(regex_test_pattern)", "$(mp_share_config_file)"), 1)
+        );
```

Comment:
```suggestion
   "$(mp_share_config_file)"
      edit_line => change_cli_rest_server_url_port,
      if => and(
        fileexists("$(mp_share_config_file)"),
        islessthan(countlinesmatching("$(regex_test_pattern)", "$(mp_share_config_file)"), 1)
        );
        
    "$(mp_config_file)"
      edit_line => change_cli_rest_server_url_port,
      if => and(
        fileexists("$(mp_config_file)"),
        islessthan(countlinesmatching("$(regex_test_pattern)", "$(mp_config_file)"), 1)
        );
```

I would target the share file first to avoid potential flip flop where config file is edited, agent runs then you fix share but the run fixes the config file back until the next agent run.